src/components/NWCEditor.tsx

```diff
@@ -0,0 +1,517 @@
+        case "Year":
+            return 3;
+    }
+}
+
+export function NWCEditor(props: {
+    initialProfileIndex?: number;
+    initialNWA?: string;
+    onSave: (indexToOpen?: number, nwcUriForCallback?: string) => Promise<void>;
+}) {
+    const [state] = useMegaStore();
+    const i18n = useI18n();
+
+    const nwa = createMemo(() => parseNWA(props.initialNWA));
+
+    const formMode = createMemo(() => {
+        const mode: "createnwa" | "createnwc" | "editnwc" = nwa()
+            ? "createnwa"
+            : props.initialProfileIndex
+            ? "editnwc"
+            : "createnwc";
+        return mode;
+    });
+
+    // NWA HANDLING SECTION
+
+    // for "createNwa" we need to parse the nwa and fetch the nostr profile if applicable
+    const [nostrProfile] = createResource(
+        () => nwa()?.identity,
+        fetchNostrProfile
+    );
+
+    const name = createMemo(() => {
+        if (!nostrProfile.latest) return;
+        const parsed = JSON.parse(nostrProfile.latest.content);
+        const name = parsed.display_name || parsed.name;
+        return name;
+    });
+
+    const image = createMemo(() => {
+        if (!nostrProfile.latest) return;
+        const parsed = JSON.parse(nostrProfile.latest.content);
+        const image_url = parsed.picture;
+        return image_url;
+    });
+
+    const parsedBudget = createMemo(() => {
+        if (!nwa()?.budget) return;
+        const [amount, interval] = nwa()!.budget!.split("/");
+        return {
+            amount,
+            interval: mapNwaInterval(interval)
+        };
+    });
+
+    async function createNwa(f: BudgetForm) {
+        if (!f.nwaString) throw new Error("We lost the NWA string!");
+        try {
+            await state.mutiny_wallet?.approve_nostr_wallet_auth(
+                f.connection_name || "Nostr Wallet Auth",
+                // can we do better than ! here?
+                f.nwaString
+            );
+        } catch (e) {
+            console.error(e);
+        } finally {
+            props.onSave();
+        }
+    }
+    // END NWA HANDLING SECTION
+
+    // REGULAR NWC STUFF
+    // If the profile has a label we can fetch the contact for showing the profile image
+
+    const nwcProfileFetcher: ResourceFetcher<
+        number,
+        NwcProfile | undefined
+    > = async (index, _last) => {
+        console.log("fetching nwc profile", index);
+        if (!index) return undefined;
+
+        try {
+            const profile: NwcProfile | undefined =
+                await state.mutiny_wallet?.get_nwc_profile(index);
+            console.log(profile);
+            return profile;
+        } catch (e) {
+            console.error(e);
+            return undefined;
+        }
+    };
+
+    const [profile] = createResource(
+        props.initialProfileIndex,
+        nwcProfileFetcher
+    );
+
+    // TODO: this should get the contact so we can get the image, but not getting a contact tagged on the nwc right now
```

Comment:
it looks like this is done

Reply:
not getting the image url in the contact response, we can revisit